bgpd/bgp_mpath.c

```diff
@@ -605,31 +605,44 @@ void bgp_path_info_mpath_update(struct bgp *bgp, struct bgp_dest *dest,
 		if (mp_node && (listgetdata(mp_node) == cur_mpath)) {
 			list_delete_node(mp_list, mp_node);
 			bgp_path_info_mpath_dequeue(cur_mpath);
-			if ((mpath_count < maxpaths)
-			    && prev_mpath
-			    && bgp_path_info_nexthop_cmp(prev_mpath,
-							 cur_mpath)) {
+			if ((mpath_count < maxpaths) && prev_mpath) {
+				mpath_count++;
+				if (bgp_path_info_nexthop_cmp(prev_mpath,
+							      cur_mpath)) {
+					if (ecommunity_linkbw_present(
+						    bgp_attr_get_ecommunity(
+							    cur_mpath->attr),
+						    &bwval) ||
+					    ecommunity_linkbw_present(
+						    bgp_attr_get_ipv6_ecommunity(
+							    cur_mpath->attr),
+						    &bwval))
+						cum_bw += bwval;
+					else
+						all_paths_lb = false;
+					if (debug) {
+						bgp_path_info_path_with_addpath_rx_str(
+							cur_mpath, path_buf,
+							sizeof(path_buf));
+						zlog_debug("%pBD: %s is still multipath, cur count %d",
+							   dest, path_buf,
+							   mpath_count);
+					}
+				} else {
+					if (debug) {
+						bgp_path_info_path_with_addpath_rx_str(
+							new_mpath, path_buf,
+							sizeof(path_buf));
+						zlog_debug("%pBD: nexthop equal, however add mpath %s nexthop %pI4, cur count %d",
```

Comment:
Is this worth printing only attr->nexthop? What if mp_nexthop_global_in, mp_nexthop_global, mp_nexthop_local exists?

Reply:
the spec you propose is not clear. adding to this, there are more places in the function , where mp_nexthop... could be displayed. I prefer doing it on a separate commit, or on a separate pull request.